deps.ml

```diff
@@ -0,0 +1,185 @@
+open Types
+open Dune_items
+
+type t = Dir_set.t Libraries.t
+
+let dune_describe_external_lib_deps () = Bos.Cmd.(v "dune" % "describe" % "external-lib-deps")
+
+let dune_describe_entries () = Bos.Cmd.(v "dune" % "describe" % "package-entries")
+
+let describe_external_lib_deps =
+  Lazy.from_fun (fun _ ->
+      sexp @@ dune_describe_external_lib_deps ()
+      |> Describe_external_lib.describe_extern_of_sexp)
+
+let describe_bin_of_entries =
+  Lazy.from_fun (fun _ ->
+      sexp @@ dune_describe_entries ()
+      |> Describe_entries.entries_of_sexp
+      |> Describe_entries.items_bin_of_entries)
+
+let has_dune_subproject = function
+  | "." | "" -> false
+  | dir -> Sys.file_exists (Filename.concat dir "dune-project")
+
+let rec should_use_dir ~dir_types path =
+  match Hashtbl.find_opt dir_types path with
+  | Some x -> x
+  | None ->
+    let r =
+      match Astring.String.cut ~sep:"/" ~rev:true path with
```

Comment:
Won't work on Windows, the file separator isn't `/`. Consider using [`Fpath`](https://erratique.ch/software/fpath/doc/Fpath/index.html).